iroha_crypto/src/lib.rs

```diff
@@ -64,48 +96,58 @@ pub struct KeyPair {
 }
 
 impl KeyPair {
+    /// Generates a pair of Public and Private key with `Algorithm::default()` selected as generation algorithm.
+    /// Returns `Err(String)` with error message if failed.
+    pub fn generate() -> Result<Self, String> {
+        Self::generate_with_algorithm(Algorithm::default())
+    }
+
     /// Generates a pair of Public and Private key with the corresponding `KeyGenOption`.
     /// Returns `Err(String)` with error message if failed.
     pub fn generate_with_option(key_gen_option: KeyGenOption) -> Result<Self, String> {
-        let (public_key, ursa_private_key) = Ed25519Sha512
-            .keypair(Some(key_gen_option.try_into()?))
-            .map_err(|e| format!("Failed to generate Ed25519Sha512 key pair: {}", e))?;
-        let public_key: [u8; 32] = public_key[..]
-            .try_into()
-            .map_err(|e| format!("Public key should be [u8;32]: {}", e))?;
-        let mut private_key = [0; 64];
-        private_key.copy_from_slice(ursa_private_key.as_ref());
+        Self::generate_with_algorithm_and_option(Algorithm::default(), key_gen_option)
+    }
+
+    /// Generates a pair of Public and Private key.
+    /// Returns `Err(String)` with error message if failed.
+    pub fn generate_with_algorithm(algorithm: Algorithm) -> Result<Self, String> {
+        let (public_key, private_key) = match algorithm {
+            Algorithm::Ed25519 => Ed25519Sha512.keypair(None),
+            Algorithm::Secp256k1 => EcdsaSecp256k1Sha256::new().keypair(None),
+        }
+        .map_err(|e| format!("Failed to generate key pair: {}", e))?;
```

Comment:
I think we can work with builder pattern here and make it from `KeyPairGenerationOption`:
```
struct KeyPairGenerationOption {
  seed: Option<...>,
  private_key: Option<...>,
  algorithm: Algorithm::default(),
}
```